### PR TITLE
python/apigen: Fix support for pybind11 overloaded functions

### PR DESCRIPTION
**Pull request by jbms**
This was accidentally broken by
https://github.com/jbms/sphinx-immaterial/pull/357.

### CODE REVIEW

**Review by 2bndy5 — COMMENTED** [review]
Its odd that the added test for this uses a pure python. Is it safe to assume pybind11 will generate bindings the same way in the future?

**Comment by jbms** [comment]
Okay, I changed the test to use actual pybind11.

File: tests/python_apigen_test_modules/pybind11_overloaded.py
Comment (2bndy5) [inline tests/python_apigen_test_modules/pybind11_overloaded.py:1]: Is this file still needed?

Comment (jbms) [reply tests/python_apigen_test_modules/pybind11_overloaded.py:1]: Removed.

**Review by 2bndy5 — APPROVED** [review]
